### PR TITLE
read featuregates from openshift/api

### PR DESCRIPTION
**Pull request by deads2k**
https://github.com/openshift/api/pull/1567 is needed first, but this is how it would be used.

/hold until 4.15

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
LGTM pending vendor changes

Are there any changes needed to the CCO render executed by the installer to go along with this?

**Comment by deads2k** [comment]
> write-available-featuresets: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by write-available-featuresets)

> write-available-featuresets: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by write-available-featuresets)

from the init container for openshift/api.  Not sure what I did wrong there.

File: manifests/0000_10_config-operator_07_deployment.yaml [hunk 1]
```diff
@@ -36,6 +36,26 @@ spec:
...
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop: ["ALL"]
+          terminationMessagePolicy: FallbackToLogsOnError
+          image: quay.io/openshift/origin-cluster-config-api:v4.0
+          imagePullPolicy: IfNotPresent
+          command:
+            - write-available-featuresets
+            - asset-output-dir=/available-featuregates
```
Comment (JoelSpeed) [inline manifests/0000_10_config-operator_07_deployment.yaml:50]: Do you want to use an empty dir for this? And use that to share the manifests between the two containers?

Currently the E2E is failing on unable to `mkdir`

**Comment by deads2k** [comment]
Looks like it works once we match the openshift/api.

This means we *could* merge, but we'd end up jamming openshift/api PRs until we managed to also complete https://github.com/openshift/installer/pull/7666, so we should wait.

**Comment by deads2k** [comment]
some infrastructure appears down

/retest

**Comment by JoelSpeed** [comment]
Failure in serial looks to be unrelated, will run once more to see if it goes green
/retest

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Comment by deads2k** [comment]
https://github.com/openshift/installer/pull/7666 switching the installer rendering to use openshift/api has tested green and is ready to make the switch

https://github.com/openshift/api/pull/1666 and https://github.com/openshift/cluster-config-operator/pull/379 have clusterbot tested together fine. They hand off the CVO management of CRDs (and others) from cluster-config-operator to api image.

https://github.com/openshift/cluster-config-operator/pull/349 switches the featuregate setting from using the vendored copy from to using openshift/api.  It tested green until it openshift/api updated without a (yet) merged openshift/cluster-config-operator update, so the machine-config-operator mismatched until they rematch. But in combination with the rendering change linked above they will match ok.

Manually merging to get a batch.

**Pull request merged**